### PR TITLE
Detect bad categories

### PR DESCRIPTION
**Pull request by fejta**
/assign @cjwagner 

https://github.com/kubernetes/test-infra/pull/6733#issuecomment-364265342

### CODE REVIEW

File: prow/plugins/cat/cat.go [hunk 1]
```diff
@@ -201,5 +201,10 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, c
 		return gc.CreateComment(org, repo, number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, resp))
 	}
 
+	if category != "" {
```
Comment (BenTheElder) [inline prow/plugins/cat/cat.go]: cant we also hit this if we just fail to reach the cat api 3 times?
could hoist err out of the loop and use `if err == nil && category != ""`

/shrug

Comment (fejta) [reply prow/plugins/cat/cat.go]: err isn't defined in this scope

Comment (BenTheElder) [reply prow/plugins/cat/cat.go]: yes, but we could increase the scope for err above
/shrug

**Comment by BenTheElder** [comment]
Fix the unit test? CatAPI being down edge case can be handled in a follow up 🤷‍♂️

**Comment by fejta** [comment]
Done

File: prow/plugins/cat/cat.go (hunk 1 above)
Comment (fejta) [reply prow/plugins/cat/cat.go]: Done

File: prow/plugins/cat/cat.go [hunk 2]
```diff
@@ -201,5 +201,15 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, c
...
 	}
 
+	var msg string
+	if category != "" {
+		msg = "Bad category. Please see http://thecatapi.com/api/categories/list"
+	} else {
+		msg = "http://thecatapi.com appears to be down"
+	}
+	if err := gc.CreateComment(org, repo, number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg)); err != nil {
+		log.WithError(err).Println("Failed to leave comment")
```
Comment (cjwagner) [inline prow/plugins/cat/cat.go]: s/Println/Error/

This also should be fixed above.

**Comment by fejta** [comment]
okay now it comments whenever thecatapi is sad

File: prow/plugins/cat/cat.go (hunk 2 above)
Comment (cjwagner) [reply prow/plugins/cat/cat.go]: Without this we can't filter for errors in the logs with `level=error`.

Comment (fejta) [reply prow/plugins/cat/cat.go]: Done
